code/modules/mob/living/carbon/human/life.dm

```diff
@@ -319,8 +319,16 @@
 	// Tissues die without blood circulation
 	adjustBruteLoss(2)
 
-
-
+/mob/living/carbon/human/handle_thirst()
+	..()
+	if(thirst >= THIRST_LEVEL_OVERHYDRATED)
+		if(prob(5) && w_uniform)
+			Stun(4 SECONDS)
+			visible_message("[src] pees [p_their()] pants!")
```

Comment:
Just all of my why....

Reply:
got any better ideas

Reply:
people pee from too much water not puke

Reply:
Not adding bathroom mechanics to /tg/.
There are plenty of meme stations you can go contribute to.

Reply:
Thats not a real suggestion

Reply:
@chesse20 people either vomit water if they drink enough to stress their system, or they simply flat die from rupturing their stomach. You only urinate after your body finished extracting fluids and resources from enough stuff that passes through your small intestine that it needs to get rid of waste buildup.

Also, adding bathroom mechanics to space station is fucking ludicrous and nobody will stand for it.

Reply:
they pee themselves and then people laugh at them. Theres no pee object and you can't pee on objects or other people -_-

Reply:
Make them slow down or something untill they go to the toilet

Reply:
people would get mad if they had to go the bathroom

Reply:
Actually, if you drink too much to upset the osmotic pressure, you can make your braincells bloat up with all the water and go pop, causing braindamage and coma. However, you'd need to take stuff that inhibits the urge to pee normally, or so I've been told.